torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,341 @@
+#     c. If an application program gives you access to functionality of the Software for development of application
+#        programs, reusable components or other software components (e.g. an application that is a scripting wrapper),
+#        usage of the application program is considered to be usage of the Software and is thus bound by this license.
+#     d. If the items are not available to the general public, and the initial developer of the Software requests a copy
+#        of the items, then you must supply one.
+#  7. Users must cite the authors of the Software upon publication of results obtained through the use of original or
+#     modified versions of the Software by referring to the following publication:
+#     P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”, submitted to WMT
+#     2019.
+#  8. In no event shall the initial developers or copyright holders be liable for any damages whatsoever, including -
+#     but not restricted to - lost revenue or profits or other direct, indirect, special, incidental or consequential
+#     damages, even if they have been advised of the possibility of such damages, except to the extent invariable law,
+#     if any, provides otherwise.
+#  9. You assume all risks concerning the quality or the effects of the SOFTWARE and its use. If the SOFTWARE is
+#     defective, you will bear the costs of all required services, corrections or repairs.
+#  10. This license has the binding value of a contract.
+#  11. The present license and its effects are subject to German law and the competent German Courts.
+#
+# The Software and this license document are provided "AS IS" with NO EXPLICIT OR IMPLICIT WARRANTY OF ANY KIND,
+# INCLUDING WARRANTY OF DESIGN, ADAPTION, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+
+import re
+import unicodedata
+from math import inf
+from typing import List, Tuple, Union
+
+from torch import Tensor, tensor
+from typing_extensions import Literal
+
+
+def _distance_between_words(reference_word: str, hypothesis_word: str) -> int:
+    """Distance measure used for substitutions/identity operation Copied from
+    https://github.com/rwth-i6/ExtendedEditDistance/blob/master/EED.py.
+
+    Args:
+        reference_word: reference word string
+        hypothesis_word: hypothesis word string
+
+    Returns:
+        0 for match, 1 for no match
+    """
+    return int(reference_word != hypothesis_word)
+
+
+def _eed_function(
+    hyp: List[str],
+    ref: List[str],
+    alpha: float = 2.0,  # optimal jump penalty
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+    rho: float = 0.3,  # coverage cost
+) -> float:
+    """Computes extended edit distance score for two strings: hypotheses and references. Copied from
+    https://github.com/rwth-i6/ExtendedEditDistance/blob/master/EED.py.
+
+    Args:
+        hyp: Transcription to score as a string
+        ref: Reference for input as a string
+
```

Comment:
please add arg description for the rest
```suggestion
        ref: Reference for input as a string
        alpha: ...
        deletion: ...
        insertaion: ...
        rho: ...

```

Reply:
Hi, thanks for taking a look at my code! Just did this